docs/doxygen/pages.dox

```diff
@@ -12,41 +12,180 @@ of this library include:
+optimized for resource constrained devices and does not dynamically allocate any
+memory.
+
+@section http_transport_interface_blurb Transport Interface
+
+The HTTP client library relies on a transport interface to decouple itself from
+the user platform. The transport interface allows the HTTP client library to send
+and receive data over the user's transport layer. The user platform must implement
+a @ref TransportInterface_t to use in @ref HTTPClient_Send.
+
+@see The [transport interface documentation](@ref http_transport_interface) for more
+information.
+
+@section http_request_serialization Building an HTTP Request
+
+The HTTP client library provides the following API to serialize request headers.
+- @ref HTTPClient_InitializeRequestHeaders
+- @ref HTTPClient_AddHeader
+- @ref HTTPClient_AddRangeHeader
+
+An application is expected to create and populate a @ref HTTPRequestInfo_t and
+set a buffer to use for the headers in @ref HTTPRequestHeaders_t.pBuffer. The
+application may first call @ref HTTPClient_InitializeRequestHeaders to populate
+the @ref HTTPRequestHeaders_t with the method, the path, and the host. The
+HTTP request will be serialized to following when HTTPRequestInfo_t.reqFlags is
+zero.
+
+@code
+<METHOD> <PATH> HTTP/1.1\r\n
+User-Agent: <MY-PLATFORM>\r\n
+Host: <SERVER-URL>\r\n\r\n
+@endcode
+
+When the @ref HTTPRequestInfo_t.reqFlags has @ref HTTP_REQUEST_KEEP_ALIVE_FLAG
+set, then the HTTP request will be serialized to the following:
+
+@code
+<METHOD> <PATH> HTTP/1.1\r\n
+User-Agent: <MY-PLATFORM>\r\n
+Host: <SERVER-URL>\r\n
+Connection: keep-alive\r\n\r\n
+@endcode
 
-[<FIXME: Some useful design information.>]
+The user application may add more headers using @ref HTTPClient_AddHeader or
+@ref HTTPClient_AddRangeHeader. New headers will be appended to the end of the
+existing headers. Please see the following example:
+
+@code
+<METHOD> <PATH> HTTP/1.1\r\n
+User-Agent: <MY-PLATFORM>\r\n
+Host: <SERVER-URL>\r\n
+Connection: keep-alive\r\n
+Another-Header1: another-value1\r\n
+Another-Header2: another-value2\r\n
+Another-Header3: another-value3\r\n\r\n
+@endcode
+
+The request body is passed into the @ref HTTPClient_Send function when the
+request is ready to be sent.
+
+@section http_range_support HTTP Range Requests and Partial Content Responses
+
+This library strongly encourages the use of Range Requests to download large
+files. Large is defined here to be a file whose total size cannot fit into the
+space currently available in RAM. By downloading a large file using range
+requests the user application can spend time processing that part of the file
+(for example writing to flash), then request the next part of the file. If the
+user application were to request the entire file at once and process it in
+sections from the network, the system is at a high risk for dropping packets.
+Dropped packets cause retransmissions and with many there can be a negative
```

Comment:
```suggestion
Dropped packets cause retransmissions and with many, there can be a negative
```

Reply:
Thanks I need the comma before the with also